server/szurubooru/func/posts.py

```diff
@@ -751,6 +803,8 @@ def transfer_flags(source_post_id: int, target_post_id: int) -> None:
     if replace_content:
         content = files.get(get_post_content_path(source_post))
         transfer_flags(source_post.post_id, target_post.post_id)
+        purge_post_signature(source_post)
+        purge_post_signature(target_post)
```

Comment:
If it's about merging posts, maybe we could just reassign `.signature` objects rather than purging them?

Reply:
I tried various combinations of this and constantly run into issues with SQLAlchemy's 'ON DELETE CASCADE' implementation.

It sucks that we have to regenerate the signature, but I don't think users are merging often enough for this to be a major setback.

Reply:
What if you flush inbetween?
Also that's fine but I see that you're only removing the signatures of both posts but not regenerating them, which means it'll stay null until the server gets restarted and repopulates missing hashes. Or am I missing something?

Reply:
Flushing doesn't seem to help. I wouldn't be surprised to find out it's a bug in SQLAlchemy, cause signatures cascade delete on the `tests/model/test_post.py` unit test just fine.

The call to `update_post_content` just below will regenerate the hashes.